### PR TITLE
[AIRFLOW-450] example dag "example_http_operator" compatible issue with Python 3

### PR DESCRIPTION
**Pull request by wjo1212**
fix below issue:
- _https://issues.apache.org/jira/browse/AIRFLOW-450_

Testing Done:
- airflow run example_http_operator http_sensor_check 2016-08-04
- tests passed

### CODE REVIEW

**Comment by msumit** [comment]
Please, follow 50/72 rule for commit messages. Also PR title should be of [AIRFLOW-XXX] format. Please see contribution guidelines at https://cwiki.apache.org/confluence/display/AIRFLOW/Contributors%27+Guide

**Comment by msumit** [comment]
changes lgtm.

File: airflow/example_dags/example_http_operator.py [hunk 1]
```diff
@@ -83,7 +83,7 @@
     http_conn_id='http_default',
     endpoint='',
     params={},
-    response_check=lambda response: True if "Google" in response.content else False,
+    response_check=lambda response: True if b"Google" in response.content else False,
```
Comment (ldct) [inline airflow/example_dags/example_http_operator.py:86]: can we change this to just `lambda response: b"Google" in response.content`?

Comment (gwax) [reply airflow/example_dags/example_http_operator.py:86]: Could we avoid bytes altogether?

``` python
lambda response: 'Google' in response.content.decode('utf-8')
```

Comment (r39132) [reply airflow/example_dags/example_http_operator.py:86]: @gwax I tried your example above (on 2.7) and it did not work:

```
[2016-11-03 23:38:49,798] {models.py:1271} INFO - Executing <Task(HttpSensor): http_sensor_check> on 2016-10-27 00:00:00
[2016-11-03 23:38:49,814] {sensors.py:609} INFO - Poking: 
[2016-11-03 23:38:49,818] {base_hook.py:67} INFO - Using connection to: https://www.google.com/
[2016-11-03 23:38:49,821] {http_hook.py:77} INFO - Sending 'GET' to url: https://www.google.com/
[2016-11-03 23:38:49,823] {connectionpool.py:805} INFO - Starting new HTTPS connection (1): www.google.com
[2016-11-03 23:38:50,146] {models.py:1340} ERROR - 'utf8' codec can't decode byte 0xa0 in position 8860: invalid start byte
Traceback (most recent call last):
  File "/Users/siddharth/Projects/airflow/airflow/models.py", line 1297, in run
    result = task_copy.execute(context=context)
  File "/Users/siddharth/Projects/airflow/airflow/operators/sensors.py", line 73, in execute
    while not self.poke(context):
  File "/Users/siddharth/Projects/airflow/airflow/operators/sensors.py", line 617, in poke
    return self.response_check(response)
  File "/Users/siddharth/Projects/airflow/my_dags/test_http.py", line 73, in <lambda>
    response_check=lambda response: 'Google' in response.content.decode('utf-8'),
  File "/Users/siddharth/Projects/airflow/venv/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xa0 in position 8860: invalid start byte
[2016-11-03 23:38:50,149] {models.py:1356} INFO - Marking task as UP_FOR_RETRY
[2016-11-03 23:38:50,170] {models.py:1385} ERROR - 'utf8' codec can't decode byte 0xa0 in position 8860: invalid start byte

```

Comment (gwax) [reply airflow/example_dags/example_http_operator.py:86]: My bad, I didn't fully understand the details of the problem. Digging into it, we can use the requests library's response object to handle the decoding:

``` python
lambda response: 'Google' in response.text
```

should do what we want

**Comment by r39132** [comment]
I'm closing this PR because though this fix might address the sensor, it doesn't address the fact that  a 404 is received on the second task in this example DAG. The example DAG is essentially broken. I welcome a fix to this example that brings it back to a working form.

@gtoonstra Is there another sample HTTP endpoint we can use? 

I'm getting a 404 when running this example. 

```
[2016-11-12 13:02:40,524] {models.py:168} INFO - Filling up the DagBag from /Users/siddharth/Projects/airflow/my_dags/test_http.py
[2016-11-12 13:02:41,462] {models.py:168} INFO - Filling up the DagBag from /Users/siddharth/Projects/airflow/my_dags/test_http.py
[2016-11-12 13:02:41,535] {models.py:1059} INFO - Dependencies all met for <TaskInstance: example_http_operator_sid_v2.post_op 2016-11-07 00:00:00 [queued]>
[2016-11-12 13:02:41,543] {models.py:1059} INFO - Dependencies all met for <TaskInstance: example_http_operator_sid_v2.post_op 2016-11-07 00:00:00 [queued]>
[2016-11-12 13:02:41,543] {models.py:1248} INFO - 
--------------------------------------------------------------------------------
Starting attempt 1 of 2
--------------------------------------------------------------------------------

[2016-11-12 13:02:41,551] {models.py:1271} INFO - Executing <Task(SimpleHttpOperator): post_op> on 2016-11-07 00:00:00
[2016-11-12 13:02:41,568] {http_operator.py:78} INFO - Calling HTTP method
[2016-11-12 13:02:41,572] {base_hook.py:67} INFO - Using connection to: https://www.google.com/
[2016-11-12 13:02:41,576] {http_hook.py:77} INFO - Sending 'POST' to url: https://www.google.com/api/v1.0/nodes
[2016-11-12 13:02:41,576] {connectionpool.py:805} INFO - Starting new HTTPS connection (1): www.google.com
[2016-11-12 13:02:41,680] {http_hook.py:102} ERROR - HTTP error: Not Found
[2016-11-12 13:02:41,680] {http_hook.py:108} ERROR - <!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Not Found)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL <code>/api/v1.0/nodes</code> was not found on this server.  <ins>That’s all we know.</ins>

[2016-11-12 13:02:41,680] {models.py:1340} ERROR - 404:Not Found
Traceback (most recent call last):
  File "/Users/siddharth/Projects/airflow/airflow/models.py", line 1297, in run
    result = task_copy.execute(context=context)
  File "/Users/siddharth/Projects/airflow/airflow/operators/http_operator.py", line 82, in execute
    self.extra_options)
  File "/Users/siddharth/Projects/airflow/airflow/hooks/http_hook.py", line 78, in run
    return self.run_and_check(session, prepped_request, extra_options)
  File "/Users/siddharth/Projects/airflow/airflow/hooks/http_hook.py", line 109, in run_and_check
    raise AirflowException(str(response.status_code)+":"+response.reason)
AirflowException: 404:Not Found
[2016-11-12 13:02:41,683] {models.py:1356} INFO - Marking task as UP_FOR_RETRY
[2016-11-12 13:02:41,700] {models.py:1385} ERROR - 404:Not Found
[2016-11-14 07:39:40,263] {models.py:168} INFO - Filling up the DagBag from /Users/siddharth/Projects/airflow/my_dags/test_http.py
[2016-11-14 07:39:41,421] {models.py:168} INFO - Filling up the DagBag from /Users/siddharth/Projects/airflow/my_dags/test_http.py
[2016-11-14 07:39:41,530] {models.py:1059} INFO - Dependencies all met for <TaskInstance: example_http_operator_sid_v2.post_op 2016-11-07 00:00:00 [queued]>
[2016-11-14 07:39:41,546] {models.py:1059} INFO - Dependencies all met for <TaskInstance: example_http_operator_sid_v2.post_op 2016-11-07 00:00:00 [queued]>
[2016-11-14 07:39:41,546] {models.py:1248} INFO - 
```

The example example_http_operator is currently broken so this fix is not super valid anyway!

**Comment by jonathanunderwood** [comment]
Am confused by the commit that closed this PR - it contains no code changes?

This does need fixing, as currently, with 1.8.1 `airflow initdb` is broken on a fresh install with Python 3.5:

```
$ airflow initdb
[2017-06-17 20:00:03,284] {__init__.py:57} INFO - Using executor SequentialExecutor
DB: sqlite:////opt/airflow/airflow/airflow.db
[2017-06-17 20:00:03,615] {db.py:287} INFO - Creating tables
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> e3a246e0dc1, current schema
INFO  [alembic.runtime.migration] Running upgrade e3a246e0dc1 -> 1507a7289a2f, create is_encrypted
/opt/airflow/lib64/python3.5/site-packages/alembic/util/messaging.py:69: UserWarning: Skipping unsupported ALTER for creation of implicit constraint
  warnings.warn(msg)
INFO  [alembic.runtime.migration] Running upgrade 1507a7289a2f -> 13eb55f81627, maintain history for compatibility with earlier migrations
INFO  [alembic.runtime.migration] Running upgrade 13eb55f81627 -> 338e90f54d61, More logging into task_isntance
INFO  [alembic.runtime.migration] Running upgrade 338e90f54d61 -> 52d714495f0, job_id indices
INFO  [alembic.runtime.migration] Running upgrade 52d714495f0 -> 502898887f84, Adding extra to Log
INFO  [alembic.runtime.migration] Running upgrade 502898887f84 -> 1b38cef5b76e, add dagrun
INFO  [alembic.runtime.migration] Running upgrade 1b38cef5b76e -> 2e541a1dcfed, task_duration
INFO  [alembic.runtime.migration] Running upgrade 2e541a1dcfed -> 40e67319e3a9, dagrun_config
INFO  [alembic.runtime.migration] Running upgrade 40e67319e3a9 -> 561833c1c74b, add password column to user
INFO  [alembic.runtime.migration] Running upgrade 561833c1c74b -> 4446e08588, dagrun start end
INFO  [alembic.runtime.migration] Running upgrade 4446e08588 -> bbc73705a13e, Add notification_sent column to sla_miss
INFO  [alembic.runtime.migration] Running upgrade bbc73705a13e -> bba5a7cfc896, Add a column to track the encryption state of the 'Extra' field in connection
INFO  [alembic.runtime.migration] Running upgrade bba5a7cfc896 -> 1968acfc09e3, add is_encrypted column to variable table
INFO  [alembic.runtime.migration] Running upgrade 1968acfc09e3 -> 2e82aab8ef20, rename user table
INFO  [alembic.runtime.migration] Running upgrade 2e82aab8ef20 -> 211e584da130, add TI state index
INFO  [alembic.runtime.migration] Running upgrade 211e584da130 -> 64de9cddf6c9, add task fails journal table
INFO  [alembic.runtime.migration] Running upgrade 64de9cddf6c9 -> f2ca10b85618, add dag_stats table
INFO  [alembic.runtime.migration] Running upgrade f2ca10b85618 -> 4addfa1236f1, Add fractional seconds to mysql tables
INFO  [alembic.runtime.migration] Running upgrade 4addfa1236f1 -> 8504051e801b, xcom dag task indices
INFO  [alembic.runtime.migration] Running upgrade 8504051e801b -> 5e7d17757c7a, add pid field to TaskInstance
INFO  [alembic.runtime.migration] Running upgrade 5e7d17757c7a -> 127d2bf2dfa7, Add dag_id/state index on dag_run table
ERROR [airflow.models.DagBag] Failed to import: /opt/airflow/lib64/python3.5/site-packages/airflow/example_dags/example_http_operator.py
Traceback (most recent call last):
  File "/opt/airflow/lib64/python3.5/site-packages/airflow/models.py", line 264, in process_file
    m = imp.load_source(mod_name, filepath)
  File "/opt/rh/rh-python35/root/usr/lib64/python3.5/imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 693, in _load
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 662, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/opt/airflow/lib64/python3.5/site-packages/airflow/example_dags/example_http_operator.py", line 20, in <module>
    from airflow.operators.sensors import HttpSensor
  File "/opt/airflow/lib64/python3.5/site-packages/airflow/operators/sensors.py", line 32, in <module>
    from airflow.hooks.hdfs_hook import HDFSHook
  File "/opt/airflow/lib64/python3.5/site-packages/airflow/hooks/hdfs_hook.py", line 20, in <module>
    from snakebite.client import Client, HAClient, Namenode, AutoConfigClient
  File "/opt/airflow/lib64/python3.5/site-packages/snakebite/client.py", line 1473
    baseTime = min(time * (1L << retries), cap);
                            ^
SyntaxError: invalid syntax
Done.
```

**Comment by jonathanunderwood** [comment]
Reported here: https://issues.apache.org/jira/browse/AIRFLOW-1316

**Pull request closed without merging**